maven-wrapper-distribution/src/resources/mvnw

```diff
@@ -193,42 +193,48 @@ else
     else
       wrapperUrl="https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/@@project.version@@/maven-wrapper-@@project.version@@.jar"
     fi
-    while IFS="=" read key value; do
+    while IFS="=" read -r key value; do
       case "$key" in (wrapperUrl) wrapperUrl="$value"; break ;;
       esac
     done < "$MAVEN_PROJECTBASEDIR/.mvn/wrapper/maven-wrapper.properties"
     log "Downloading from: $wrapperUrl"
 
     if $cygwin; then
-      wrapperJarPath=`cygpath --path --windows "$wrapperJarPath"`
+      wrapperJarPath=$(cygpath --path --windows "$wrapperJarPath")
     fi
 
     if command -v wget > /dev/null; then
         log "Found wget ... using wget"
+        wgetrc=0
         [ "$MVNW_VERBOSE" = true ] && QUIET="" || QUIET="--quiet"
         if [ -z "$MVNW_USERNAME" ] || [ -z "$MVNW_PASSWORD" ]; then
             wget $QUIET "$wrapperUrl" -O "$wrapperJarPath"
+            wgetrc=$?
         else
             wget $QUIET --http-user="$MVNW_USERNAME" --http-password="$MVNW_PASSWORD" "$wrapperUrl" -O "$wrapperJarPath"
+            wgetrc=$?
         fi
-        [ $? -eq 0 ] || rm -f "$wrapperJarPath"
+        [ $wgetrc -eq 0 ] || rm -f "$wrapperJarPath"
     elif command -v curl > /dev/null; then
         log "Found curl ... using curl"
+        curlrc=0
         [ "$MVNW_VERBOSE" = true ] && QUIET="" || QUIET="--silent"
         if [ -z "$MVNW_USERNAME" ] || [ -z "$MVNW_PASSWORD" ]; then
             curl $QUIET -o "$wrapperJarPath" "$wrapperUrl" -f -L
+            curlrc=$?
         else
             curl $QUIET --user "$MVNW_USERNAME:$MVNW_PASSWORD" -o "$wrapperJarPath" "$wrapperUrl" -f -L
+            curlrc=$?
         fi
-        [ $? -eq 0 ] || rm -f "$wrapperJarPath"
+        [ $curlrc -eq 0 ] || rm -f "$wrapperJarPath"
```

Comment:
instead of adding a new var, we could use sth like `if wget ...` directly. I would really remove the original line 229 though (`[ $? -eq 0]`), because it is easy to forget about it and add yet another command in between, e.g. a debugging print or echo statement which will make the check wrong.

Reply:
Do I miss something or the shell will exit if curl/wget fail ?
If so the `rm -f "$wrapperJarPath"` is useless anyway...

Reply:
```
if command -v wget > /dev/null; then
    log "Found wget ... using wget"
    [ "$MVNW_VERBOSE" = true ] && QUIET="" || QUIET="--quiet"
    { [ -z "$MVNW_USERNAME" ] || [ -z "$MVNW_PASSWORD" ] ;} \
        && USEROPT="" \
        || USEROPT="--http-user=\"$MVNW_USERNAME\" --http-password=\"$MVNW_PASSWORD\""
    wget $QUIET $USEROPT "$wrapperUrl" -O "$wrapperJarPath"
elif command -v curl > /dev/null; then
    log "Found curl ... using curl"
    [ "$MVNW_VERBOSE" = true ] && QUIET="" || QUIET="--silent"
    { [ -z "$MVNW_USERNAME" ] || [ -z "$MVNW_PASSWORD" ] ;} \
        && USEROPT="" \
        || USEROPT="--user \"$MVNW_USERNAME:$MVNW_PASSWORD\""
    curl $QUIET $USEROPT -o "$wrapperJarPath" "$wrapperUrl" -f -L
else
```

Reply:
Yes, we would need either an OR, or a trap function.
But.

The script won't exit anyway because neither `set -e` nor `set -o pipefail` is set.

Reply:
Ah, missed that.  So my first proposal should work and keep the existing behaviour.

Reply:
Done, thanks! I left out the concatination of user and password for now, but we can create a separate issue for this.